internal/pkg/agent/cmd/run.go

```diff
@@ -136,17 +139,34 @@ func run(override cfgOverrider, testingMode bool, fleetInitTimeout time.Duration
 	go service.ProcessWindowsControlEvents(stopBeat)
 
 	// detect otel
-	if runAsOtel := otel.IsOtelConfig(ctx, paths.ConfigFile()); runAsOtel {
-		return otel.Run(ctx, cancel, stop, testingMode)
+	runAsOtel := otel.IsOtelConfig(ctx, paths.ConfigFile())
+	var awaiters awaiters
+	var resErr error
+	if runAsOtel {
+		var otelStartWg sync.WaitGroup
+		otelAwaiter := make(chan struct{})
+		awaiters = append(awaiters, otelAwaiter)
+
+		otelStartWg.Add(1)
+		go func() {
+			otelStartWg.Done()
+			err := otel.Run(ctx, stop)
+			resErr = multierror.Append(resErr, err)
+
+			// close awaiter handled in run loop
+			close(otelAwaiter)
+		}()
+
+		// wait for otel to start
+		otelStartWg.Wait()
 	}
 
 	// not otel continue as usual
```

Comment:
I think we should remove this comment, now that `otel.Run` could happen concurrently with `runElasticAgent` below?